src/client/client.cpp

```diff
@@ -782,11 +782,20 @@ void Client::initLocalMapSaving(const Address &address,
 		return;
 	}
 
-	const std::string world_path = porting::path_user
-		+ DIR_DELIM + "worlds"
-		+ DIR_DELIM + "server_"
+	std::string world_path;
+#define set_world_path(hostname) \
+	world_path = porting::path_user \
+		+ DIR_DELIM + "worlds" \
+		+ DIR_DELIM + "server_" \
 		+ hostname + "_" + std::to_string(address.getPort());
 
+	set_world_path(hostname);
+	if (!fs::IsDir(world_path)) {
+		std::string hostname_escaped = hostname;
+		str_replace(hostname_escaped, ':', '_');
```

Comment:
why not just unconditionally escape?
or put the code that escapes inside an `#ifdef _WIN32`?

Reply:
`:` cannot be used in general on FAT16, FAT32 and NTFS filesystems.

Reply:
That's mostly a Windows limitation, you can create such directories or files from Linux.

Reply:
Meanwhile on Ubuntu + USB flash drive:
```
$ sudo file -s /dev/sdf
/dev/sdf: DOS/MBR boot sector, code offset 0x58+2, OEM-ID "MSDOS5.0", <snip>, sectors 4142080 (volumes > 32 MB), FAT (32 bit), sectors/FAT 4038, reserved 0x1, <snip>, unlabeled
$ LANG=en touch "te:st.txt"
touch: setting times of 'te:st.txt': No such file or directory
```

Reply:
This should be replaced on all programs, using such characters causes portability problems

Reply:
I tossed up whether to replace globally or only on Windows and I decided that other OS users will still have issues from (as mentioned): FAT filesystems, etc. The escape is left conditional only to support existing linux world downloads.